components/net/data_loader.rs

```diff
@@ -44,11 +44,13 @@ pub fn decode(url: &ServoUrl) -> Result<DecodeData, DecodeError> {
 
     let mut bytes = percent_decode(parts[1].as_bytes()).collect::<Vec<_>>();
     if is_base64 {
-        // FIXME(#2909): It’s unclear what to do with non-alphabet characters,
-        // but Acid 3 apparently depends on spaces being ignored.
+        // FIXME(#2909):
+        // infra.spec.whatwg.org/#forgiving-base64-decode does a pretty good
+        // job of telling us what to do here. Most WPT tests are working;
+        // the base64 string "ab=" in particular is not.
         bytes = bytes
             .into_iter()
-            .filter(|&b| b != b' ')
+            .filter(|&b| b != b' ' && b != b'\x0C' && b != b'\n' && b != b'\r' && b != b'\t')
```

Comment:
While it's logical to extend the existing code here, we should honour SimonSapin's request in https://github.com/servo/servo/issues/2909#issuecomment-558328602 to make use of the [data-url crate](https://docs.rs/data-url/0.1.0/data_url/) and replace all of this decoding code if possible.

We should be able to use `DataUrl::process(url.as_str())` and return the value of decode_to_vec() along with the url's mime_type() value. We'll need to add data-url as a new dependency in components/net/Cargo.toml as well.

Reply:
Will do! I've never added a Cargo dependency before, but I'll have to sooner or later and I may as well start with this one!